sources/logdog/build.rs

```diff
@@ -30,3 +39,36 @@ fn main() {
     let mut readme = File::create("README.md").unwrap();
     readme.write_all(content.as_bytes()).unwrap();
 }
+
+/// Creates a file named `variant` which is a symlink to a file with `logdog` commands for the
+/// current variant. Whatever the value of the `VARIANT` environment variable is, this function
+/// requires a file of the same name to exist in ./static, and points to it from a symlink.
+fn symlink_variant() {
+    let variant = env::var(ENV_VARIANT).unwrap_or_else(|_| {
+        eprintln!("For local builds, you must set the {} environment variable so we know which logdog commands to \
+        build. Valid values are the directories in models/src/variants/, for example 'aws-k8s-1.17'.", ENV_VARIANT);
+        process::exit(1);
+    });
+    if !PathBuf::from(format!("static/{}", variant)).is_file() {
```

Comment:
I'm a little confused about this directory; let me suggest something...

We make `sources/logdog/etc` (or `conf` or something if we hate `etc`) where we keep our variant-specific files, which we'd name `logdog.VARIANT.conf`.  We have our writable directory that we mount in via the Dockerfile, and we put it at `soures/logdog/etc/current`, and we symlink `current/logdog.conf` -> `../logdog.VARIANT.conf`.  Then I think we don't need the copies in the Dockerfile, and (imo) it turns out a little simpler.  (And we could still have `sources/logdog/etc/logdog.common.conf` or similar.)

Reply:
How's this?

```
├── .gitignore # (ignore conf/current/)
├── Cargo.toml
├── README.md
├── README.tpl
├── build.rs
├── conf
│   ├── aws-k8s.conf
│   ├── current
│   │   ├── logdog.common.conf -> ../logdog.common.conf
│   │   └── logdog.conf -> ../logdog.aws-k8s-1.17
│   ├── logdog.aws-dev.conf
│   ├── logdog.aws-ecs-1.conf
│   ├── logdog.aws-k8s-1.15.conf -> aws-k8s.conf
│   ├── logdog.aws-k8s-1.16.conf -> aws-k8s.conf
│   ├── logdog.aws-k8s-1.17.conf -> aws-k8s.conf
│   └── logdog.common.conf
└── src
    ├── commands.rs
    ├── create_tarball.rs
    ├── error.rs
    └── main.rs
```

Reply:
Only logdog itself needs to read these files, so I don't think the common file needs to be linked into the writable directory, it'd just complicate build.rs.  Looks good though. :+1:

Reply:
Oh yeah, good point, thanks.

Reply:
[afdc876](https://github.com/bottlerocket-os/bottlerocket/compare/6eb611d17506ab4aad98e5db4cd17b00bacd0839..afdc8765a9e97bb1d4436eda9b0c964fce24ab95)